### PR TITLE
- Add thresholds to allow run data to be ignored if it doesn't exceed se...

### PR DESCRIPTION
**Pull request by mdevine82**
Added in functionality referenced in staabm/xhprof.io#36

### CODE REVIEW

File: inc/prepend.php [hunk 1]
```diff
@@ -84,3 +89,40 @@ function xhprof_shutdown()
         throw $e;
     }
 }
+
+// Check profile run times and opt not to store the data if they are below
+// set thresholds.  This should reduce amount of stored data for production 
+// instances.
+function check_min_run_values($profile_data, $xhprofMainConfig){
```
Comment (staabm) [inline inc/prepend.php]: xhprof prefix is missing

File: inc/prepend.php [hunk 2]
```diff
@@ -84,3 +89,40 @@ function xhprof_shutdown()
...
+        $store_flag = false;
+    }
+    
+    $peakMemUsageExists = isset($lastElement['pmu']) && isset($xhprofMainConfig['min_peak_mem_usage']);
+    $isPeakMemUsageLow = $lastElement['pmu'] < $xhprofMainConfig['min_peak_mem_usage'];
+    if($store_flag &&  $peakMemUsageExists && $isPeakMemUsageLow){
+        $store_flag = false;
+    }
+    
+    var_dump($store_flag);
```
Comment (staabm) [inline inc/prepend.php]: Debug out

File: xhprof/includes/config.inc.sample.php [hunk 3]
```diff
@@ -5,5 +5,9 @@
     'pdo' => new PDO('mysql:dbname=your_database_name;host=localhost;charset=utf8', 'username', 'password'),
     'tmp_table_engine' => 'Memory', // MySQL Table Engine used for temporary tables
     'cache_expiration' => '60', // How many seconds a browser allowed to cache profilling results
-    'profiler_enabled' => true // Global switch to disable the profiler by default
-);
+    'profiler_enabled' => true, // Global switch to disable the profiler by default
+    'min_wall_time' => 0, //Only store runs with total wall time greater than value (microseconds)
```
Comment (staabm) [inline xhprof/includes/config.inc.sample.php]: Missing space after //

File: inc/prepend.php [hunk 4]
```diff
@@ -84,3 +89,40 @@ function xhprof_shutdown()
...
+    $lastElement = end($profile_data);
+
+    $wallTimeExists = isset($lastElement['wt']) && isset($xhprofMainConfig['min_wall_time']);
+    $isWallTimeFast = $lastElement['wt'] < $xhprofMainConfig['min_wall_time'];
+    if( $wallTimeExists && $isWallTimeFast) {
+        $store_flag = false;
+    }
+    
+    $cpuTimeExists = isset($lastElement['cpu']) && isset($xhprofMainConfig['min_cpu_time']);
+    $isCpuTimeFast = $lastElement['cpu'] < $xhprofMainConfig['min_cpu_time'];
```
Comment (staabm) [inline inc/prepend.php]: When not isset this will trigger warnings..?

**Comment by mdevine82** [comment]
BTW, I'll rebase it once it is accepted

File: inc/prepend.php [hunk 5]
```diff
@@ -84,3 +89,36 @@ function xhprof_shutdown()
...
+// Check profile run times and opt not to store the data if they are below
+// set thresholds.  This should reduce amount of stored data for production 
+// instances.
+function xhprof_check_min_run_values($profile_data, $xhprofMainConfig){
+    $store_flag = true;
+    
+    $lastElement = end($profile_data);
+
+    $wallTimeExists = isset($lastElement['wt']) && isset($xhprofMainConfig['min_wall_time']);
+
```
Comment (staabm) [inline inc/prepend.php]: delete this line and we are good to go.

**Comment by staabm** [comment]
did you test it already? does it work for you?

**Comment by mdevine82** [comment]
Yeah I tested it and all appeared to work on my side.  I'll fix those whitespace issues, was using a different editor and forgot it had a formatter setup.

**Comment by staabm** [comment]
great. will land it and tag a release after this is finished. thank you so far.

**Comment by mdevine82** [comment]
No problem, just started to get myself acclimated to the codebase.

**Comment by staabm** [comment]
its a mix of different styles created by different people :-).

**Comment by mdevine82** [comment]
BTW the code is ready if you didn't notice.

**Comment by staabm** [comment]
Thanks. Github does not notify on push, so I didnt notice.

**Comment by staabm** [comment]
Fyi: Tagged a new release

**Pull request merged**